client/service-worker-manager.js

```diff
@@ -18,30 +18,60 @@ function urlBase64UrlToUint8Array(base64UrlData) {
   return buffer;
 }
 
+/**
+ *  get service worker registration
+ */
+function getServiceworkerRegistration() {
+  return new Promise(function(resolve, reject) {
+    navigator.serviceWorker.getRegistrations().then(registration => {
+      resolve(registration);
+    });
+  });
+}
+
+/**
+ *  register service worker
+ */
+function registerServiceWorker() {
+  navigator.serviceWorker.register('sw.js').then(
+    function(serviceWorkerRegistration) {
+      console.log('Service Worker Registration Success.');
+      // Push Manager
+      if ('PushManager' in window) {
+        navigator.serviceWorker.ready.then(function () {
+          const convertedVapidKey =
+          urlBase64UrlToUint8Array(pushKey.pushVapidKeys.publicKey);
+          serviceWorkerRegistration.pushManager.subscribe({
+            userVisibleOnly: true,
+            applicationServerKey: convertedVapidKey,
+          }).then(
+            function(pushSubscription) {
+              push.registerPushSubscription(pushSubscription);
+            }, function(error) {
+              console.log(error);
+            }
+          );
+        });
+      }
+    }
+  ).catch(function(error) {
+      console.error('Service Worker Registration Fail. ', error);
+  });
+}
+
 if ('serviceWorker' in navigator) {
   window.addEventListener('load', function() {
-    navigator.serviceWorker.register('sw.js').then(
-      function(serviceWorkerRegistration) {
-        console.log('Service Worker Registration Success.');
-        // Push Manager
-        if ('PushManager' in window) {
-          navigator.serviceWorker.ready.then(function () {
-            const convertedVapidKey =
-            urlBase64UrlToUint8Array(pushKey.pushVapidKeys.publicKey);
-            serviceWorkerRegistration.pushManager.subscribe({
-              userVisibleOnly: true,
-              applicationServerKey: convertedVapidKey,
-            }).then(
-              function(pushSubscription) {
-                push.registerSubscription(pushSubscription);
-              }, function(error) {
-                console.log(error);
-              }
-            );
-          });
-        }
-      }).catch(function(error) {
-        console.error('Service Worker Registration Fail. ', error);
-      });
+    Promise.resolve()
+    .then(push.getPushPermissionStatus)
+    .then(permission => {
+      if (permission === 'granted') {
+        // TODO(jimmy) need to show popup notification for revoke
```

Comment:
Could you add ```:``` (colon) between TODO(your_id) and blah blah..
TODO(jimmy): need to show popup notification for revoke.